Ray APIs moved to private subpackage on latest master, breaking compatibility

### Describe the problem

Ray [recently moved](https://github.com/ray-project/ray/pull/14607) some of their internal APIs to an explicit `_private` subpackage. Modin calls one of these functions (`ray.utils.get_system_memory()`), leading to errors on tests that run on Ray's latest master.

### Source code / logs
[Example CI logs with failing tests](https://github.com/ray-project/xgboost_ray/pull/66/checks?check_run_id=2114689375)

```
                if object_store_memory is None:
                    # Round down to the nearest Gigabyte.
                    object_store_memory = int(
>                       0.6 * ray.utils.get_system_memory() // 10 ** 9 * 10 ** 9
                    )
E                   AttributeError: module 'ray' has no attribute 'utils'

/opt/hostedtoolcache/Python/3.6.9/x64/lib/python3.6/site-packages/modin/engines/ray/utils.py:132: AttributeError
```

<!-- Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem. -->

--- a/modin/engines/ray/utils.py
+++ b/modin/engines/ray/utils.py
@@ -133,15 +133,21 @@ def initialize_ray(
                 # want to overwrite that value if we have.
                 if object_store_memory is None:
                     # Round down to the nearest Gigabyte.
-                    mem_bytes = ray.utils.get_system_memory() // 10 ** 9 * 10 ** 9
+                    try:
+                        system_memory = ray._private.utils.get_system_memory()
+                    except AttributeError:  # Compatibility with Ray <= 1.2
+                        system_memory = ray.utils.get_system_memory()
+                    mem_bytes = system_memory // 10 ** 9 * 10 ** 9
                     # Default to 8x memory for out of core
                     object_store_memory = 8 * mem_bytes
             # In case anything failed above, we can still improve the memory for Modin.
             if object_store_memory is None:
                 # Round down to the nearest Gigabyte.
-                object_store_memory = int(
-                    0.6 * ray.utils.get_system_memory() // 10 ** 9 * 10 ** 9
-                )
+                try:
+                    system_memory = ray._private.utils.get_system_memory()
+                except AttributeError:  # Compatibility with Ray <= 1.2
+                    system_memory = ray.utils.get_system_memory()
+                object_store_memory = int(0.6 * system_memory // 10 ** 9 * 10 ** 9)
                 # If the memory pool is smaller than 2GB, just use the default in ray.
                 if object_store_memory == 0:
                     object_store_memory = None

